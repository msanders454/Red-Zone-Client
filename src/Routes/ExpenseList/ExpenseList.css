.RedZone-Budget{
    text-align: center;
    margin-left: 26%;
    margin-right: 3%;
    display: flex;
    font-size: 24px;
    font-weight: 500;
}

.state-amount{
    margin-bottom: 2%;
    margin-left: 3%;
    display: flex;
    font-size: 24px;
}

.ExpenseList__Filters {
    margin-top: 3%;
    margin-left: 3%;
    margin-right: 3%;
    display: flex;
}

.ExpenseList__Select {
    width: 30vw;
    margin: 2px;
    height: 30px;
    border-radius: 5px;
    border: 1px solid #000000;
    font-size: 0.5rem;
    padding: 1%;
}

.ExpenseList {
    text-align: center;
}


@media only screen and (min-width: 420px) {
    .ExpenseList__Select {
        font-size: 0.75rem;
    }

}

@media only screen and (min-width: 520px) {
    .ExpenseList__Select {
        font-size: 0.80rem;
    }
}

@media only screen and (min-width: 620px) {
    .ExpenseList__Select {
        font-size: 0.85rem;
        padding: 0.05%;
    }
}

@media only screen and (min-width: 720px) {
    .ExpenseList__Select {
        font-size: 0.9rem;
    }
}

@media only screen and (min-width: 784px) {
    .RedZone-Budget{
        margin-left: 36%;
    }
    .ExpenseList__Filters {
        margin-top: 0.5%;
        display: flex;
        margin-left: 25%;
    }
    .ExpenseList__Select {
        width: 20vw;
        margin: 3px;
    }
}

@media only screen and (min-width: 820px) {
    .RedZone-Budget{
        margin-left: 39%;
    }
    .ExpenseList__Select {
        height: 32px;
    }
}

@media only screen and (min-width: 1120px) {
    .RedZone-Budget{
        margin-left: 42%;
    }
    .ExpenseList__Select {
        margin: 4px;
    }
}

@media only screen and (min-width: 1220px) {
    .ExpenseList__Filters {
        margin-left: 22.5%;
    }
}

@media only screen and (min-width: 1320px) {
    .ExpenseList__Filters {
        width: 75%;
    }
    .ExpenseList__Select {
        margin: 5px;
        font-size: 1rem;
    }
    .ExpenseList__Copyright {
        font-size: 18px;
    }
}

@media only screen and (min-width: 1370px) {
    .ExpenseList__Filters {
        width: 55%;
    }
}

@media only screen and (min-width: 1420px) {
    .ExpenseList__Select {
        width: 15vw;
        margin: 10px;
    }
}