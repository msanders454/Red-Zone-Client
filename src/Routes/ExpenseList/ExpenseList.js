import React, { Component } from 'react';
import ExpenseItem from '../ExpenseItem/ExpenseItem';
import RedZoneContext from '../../RedZoneContext';
import './ExpenseList.css';

export default class ExpenseList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            expenses: props.expenses || [],
            red_zone_amount: props.red_zone_amount,
        };
    }

    static contextType = RedZoneContext;
    
    sortDateNew = () => {
        const { expenses } = this.state;
        let newerDates = [...expenses].sort(
            (a, b) => new Date(b.date) - new Date(a.date)
        );
        this.setState({
            expenses: newerDates
        });
    };

    sortDateOld = () => {
        const { expenses } = this.state;
        let olderDates = [...expenses].sort(
            (a, b) => new Date(a.date) - new Date(b.date)
        );
        this.setState({
            expenses: olderDates
        });
    };

    amountAscending = () => {
        const { expenses } = this.state;
        let ascendingList = [...expenses].sort(
            (a, b) => parseFloat(a.amount) - parseFloat(b.amount)
        );
        this.setState({
            expenses: ascendingList
        });
    };

    amountDescending = () => {
        const { expenses } = this.state;
        let descendingList = [...expenses].sort(
            (a, b) => parseFloat(b.amount) - parseFloat(a.amount)
        );
        this.setState({
            expenses: descendingList
        });
    };

    filterCategory = e => {
        const { expenses } = this.context;
        const styles = [e.target.value];
        let filteredList = [];
        if(styles.includes('AllCategories')) {
            filteredList = this.context.expenses
        } else {
            filteredList = expenses.filter(i => styles.includes(i.style));
        }
        this.setState({
            expenses: filteredList
        });
    };
    render() {
        const { expenses } = this.state;
        return (
            <div>
                <section className='RedZone-Amount'>
                <p className='RedZone-Budget'>
                <span className='RedZone__title'>RedZone Budget is: $</span>
                    <span className='state-amount'>{this.state.red_zone_amount}</span>
                </p>
                </section>
                <section className='ExpenseList__Filters'>
                    <label htmlFor='ExpenseList__Dates'></label>
                    <select
                        className='ExpenseList__Select'
                        onChange={e => {
                            if (e.target.value === 'OldToNew') {
                                this.sortDateOld();
                            } else if (e.target.value === 'NewToOld') {
                                this.sortDateNew();
                            }
                        }}
                    >
                        <option>Sort By Date</option>
                        <option value='OldToNew'>Oldest to Newest</option>
                        <option value='NewToOld'>Newest to Oldest</option>
                    </select>

                    <label htmlFor='ExpenseList_Amount'></label>
                    <select
                        className='ExpenseList__Select'
                        onChange={e => {
                            if (e.target.value === 'LowToHigh') {
                                this.amountAscending();
                            } else if (e.target.value === 'HighToLow') {
                                this.amountDescending();
                            }
                        }}
                    >
                        <option>Sort By Amount</option>
                        <option value='LowToHigh'>Lowest to Highest</option>
                        <option value='HighToLow'>Highest to Lowest</option>
                    </select>

                    <label htmlFor='ExpenseList_Category'></label>
                    <select
                        className='ExpenseList__Select'
                        onChange={this.filterCategory}
                    >
                        <option value='AllCategories'>All Categories</option>
                        <option value='Credit Cards or Loans'>Credit Cards or Loans</option>
                        <option value='Entertainment'>Entertainment</option>
                        <option value='Food'>Food</option>
                        <option value='Housing and Utilities'>Housing and Utilities</option>
                        <option value='Transportation'>Transportation</option>
                        <option value='Travel'>Travel</option>
                        <option value='Family'>Family</option>
                        <option value='Personal care and Clothing'>Personal care and Clothing</option>
                        <option value='Other'>Other</option>
                    </select>
                </section>

                <section className='ExpenseList'>
                    <ul>
                        {expenses.map(expense => (
                            <ExpenseItem
                                key={expense.id}
                                date={expense.Date}
                                amount={expense.Amount}
                                style={expense.Style}
                                description={expense.Description}
                                {...expense}
                            />
                        ))}
                    </ul>
                </section>
            </div>
        );
    }
}