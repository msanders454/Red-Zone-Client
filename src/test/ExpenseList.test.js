/* eslint-disable no-undef */

import React from "react";
import { shallow, configure } from "enzyme";
import ExpenseList from "../Routes/ExpenseList/ExpenseList";
import Adapter from "enzyme-adapter-react-16";
 
configure({ adapter: new Adapter() });
test("should render ExpenseList correctly", () => {
  const wrapper = shallow(<ExpenseList />);
  expect(wrapper).toMatchSnapshot();
});