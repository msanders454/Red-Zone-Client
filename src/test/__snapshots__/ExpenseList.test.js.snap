// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList correctly 1`] = `ShallowWrapper {}`;
